### ISSUE TITLE
StaticFile on symbolic link

### ISSUE THREAD

**Issue by gekomad** [http4s/http4s#6363]
On Linux and http4s 1.0.0-M32 this line works `StaticFile.fromPath(fs2.io.file.Path(s"/home/foo/index.html"), Some(req)).getOrElseF(NotFound())` but if the file is a symbolik link I see only 2 rows of html page on chrome, fireforx and curl. This problem there isn't on http4s 0.21.33

**Comment by armanbilge** [comment]
Thanks for reporting! Would you mind checking if the issue exists in 1.0.0-M27 and 1.0.0-M28? It may have been introduced in https://github.com/http4s/http4s/pull/5226.

**Comment by armanbilge** [comment]
Actually, if you don't mind checking v0.23.0 and v0.23.1 as well that would be really helpful.

**Comment by gekomad** [comment]
1.0.0-M28 doesn't work
1.0.0-M27 with `StaticFile.fromFile` works

**Comment by armanbilge** [comment]
@gekomad thank you for trying it, that is very helpful. I'd like to fix this but I'll need your help to reproduce it first. I wrote a test in https://github.com/http4s/http4s/commit/85b4e622a91d5f515a0d76d23cdd6421b14ff366 but it seems to work ok.

**Comment by gekomad** [comment]
this my project:

```scala
val Http4sVersion = "1.0.0-M32"

version := "0.0.1-SNAPSHOT"
scalaVersion := "2.13.8"
libraryDependencies += "org.http4s"                 %% "http4s-blaze-server"    % Http4sVersion
libraryDependencies += "org.http4s"                 %% "http4s-dsl"    % Http4sVersion
```

```scala
package com.example.quickstart

import cats.effect._
import org.http4s.blaze.server.BlazeServerBuilder
import org.http4s.{HttpRoutes, StaticFile}
import cats.effect.unsafe.implicits.global
import org.http4s.dsl.io._

object Main extends IOApp {

  val routes = HttpRoutes.of[IO] {
    case request@GET -> Root =>
          StaticFile.fromPath(fs2.io.file.Path("/home/foo/index.html"), Some(request)).getOrElseF(NotFound())
  }

  override def run(args: List[String]): IO[ExitCode] =
    BlazeServerBuilder[IO]
      .bindHttp(9000, "0.0.0.0").withHttpApp(routes.orNotFound).resource.use(_ => IO.never).start.unsafeRunSync().joinWithNever
}

```

**Comment by armanbilge** [comment]
@gekomad thanks, can you upload that into a git repo with the symlink setup as well?

**Comment by gekomad** [comment]
I created https://github.com/gekomad/http4s-test-symbolic-link
Before run test it's need create file and symlink with 
```
yes "Some text" | head -n 1000 > /tmp/myfile.txt
ln -s /tmp/myfile.txt /tmp/link.txt
```
than
```
sbt test
```

**Comment by armanbilge** [comment]
@gekomad thanks for all your help, I've PRed a fix in https://github.com/http4s/http4s/pull/6368.

**Comment by armanbilge** [comment]
@gekomad the fix is now available in `1.0-d8f71e3-SNAPSHOT`.
https://s01.oss.sonatype.org/content/repositories/snapshots/org/http4s/http4s-core_2.13/1.0-d8f71e3-SNAPSHOT/
